### PR TITLE
improve apiserver-proxy tests resilience

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Increase resilience of apiserver-proxy tests

Changes proposed in this pull request:

- Add retries to apiserver-proxy test
- Add configurable HTTP client timeout
- Refactor the test to be more readable

**Related issue(s)**
See also #3607

### CODE REVIEW

File: tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go [hunk 1]
```diff
@@ -61,45 +109,56 @@ func (p *dexIdTokenProvider) implicitFlow() (map[string]string, error) {
...
 	approvalEndpoint := loginResp.Header.Get("location")
 	if strings.Contains(approvalEndpoint, "#.*error") {
-		return nil, fmt.Errorf("approval - Redirected with error: '%s'", approvalEndpoint)
+		return "", fmt.Errorf("approval - Redirected with error: '%s'", approvalEndpoint)
 	}
+
+	return approvalEndpoint, nil
+}
+
+func (p *dexIdTokenProvider) tokenCall(approvalEndpoint string) (*url.URL, error) {
```
Comment (kubadz) [inline tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go]: Same suggestion as above.

what: receive token
how: performs a call to the approval endpoint

File: tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go [hunk 2]
```diff
@@ -44,6 +45,53 @@ func (p *dexIdTokenProvider) fetchIdToken() (string, error) {
...
+
+	if happyRun {
+		log.Println("Flow finished flawlessly")
+	}
+
+	return result, nil
+}
+
+//Performs call to the <authorize> endpoint.
+func (p *dexIdTokenProvider) authorizeCall() (string, error) {
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go]: Done.

File: tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go (hunk 1 above)
Comment (Tomasz-Smelcerz-SAP) [reply tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go]: Done.

File: tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go [hunk 3]
```diff
@@ -61,45 +109,56 @@ func (p *dexIdTokenProvider) implicitFlow() (map[string]string, error) {
...
 		if strings.Contains(loginEndpoint, "#.*error") {
-			return nil, fmt.Errorf("login - Redirected with error: '%s'", loginEndpoint)
+			return "", fmt.Errorf("login - Redirected with error: '%s'", loginEndpoint)
 		}
 	}
 
+	return loginEndpoint, nil
+}
+
+func (p *dexIdTokenProvider) loginCall(loginEndpoint string) (string, error) {
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go]: Done.

**Pull request merged**